src/librustc/ich/impls_ty.rs

```diff
@@ -176,15 +176,20 @@ impl<'a> HashStable<StableHashingContext<'a>> for mir::interpret::Allocation {
         hasher: &mut StableHasher<W>,
     ) {
         let mir::interpret::Allocation {
-            bytes, relocations, undef_mask, align, mutability,
+            relocations, align, mutability, size,
             extra: _,
+            .. /* private bytes and undef_mask */
         } = self;
```

Comment:
Hm, the reason we use a pattern here is to be alerted when new fields are added, so that we don't forget to also hash them.

@oli-obk is there any way to keep the fields private while having reliable stable hashing? Can we derive this, or move the impl to `allocation.rs`?

Reply:
`#[derive(HashStable)]` exists but I'll leave it to @oli-obk whether we can use that for `Allocation`.

Reply:
If we implement `HashStable` for `relocations` we should be able to use the derive. Or we just move this impl to `librustc/mir/interpret/allocation.rs`